src/main/java/net/atos/zac/zoeken/IndexeerService.java

```diff
@@ -233,13 +253,16 @@ private ZoekObject convertToZoekObject(
         return zoekObject;
     }
 
-    private long addToSolrIndex(final Stream<ZoekObject> zoekObjecten) {
+    private long addToSolrIndex(final Stream<ZoekObject> zoekObjecten, final boolean performCommit) {
         final List<ZoekObject> beansToBeAdded = zoekObjecten
                 .filter(Objects::nonNull)
                 .toList();
         if (CollectionUtils.isNotEmpty(beansToBeAdded)) {
             try {
                 solrClient.addBeans(beansToBeAdded);
+                if (performCommit) {
+                    solrClient.commit();
```

Comment:
Did you consider the overloads of this function? At least for the assigning/releasing tasks use case, I think you would want to set both booleans to true. Also, I think it might be a bit more flexible to have a seperate `commit` method on IndexeerService. That way, you wouldn't have to modify the existing methods of this class.

Reply:
>At least for the assigning/releasing tasks use case, I think you would want to set both booleans to true.

That's exactly what I'm doing no?

I decided not to create an extra overloaded method because I thought this was more clear. Also, when converting this code to Kotlin we can just use default function parameters which is much cleaner than Java overloaded method I think. :-)

Reply:
>Also, I think it might be a bit more flexible to have a seperate commit method on IndexeerService.

I'm not sure. Because then you would also need to add specific exception handling which is now part of the `addToSolrIndex` method.

Reply:
> > At least for the assigning/releasing tasks use case, I think you would want to set both booleans to true.
> 
> That's exactly what I'm doing no?
> 
> I decided not to create an extra overloaded method because I thought this was more clear. Also, when converting this code to Kotlin we can just use default function parameters which is much cleaner than Java overloaded method I think. :-)

This overload waits for the search index to actually be ready: `solrClient.commit(null, true, true)`

Reply:
> > Also, I think it might be a bit more flexible to have a seperate commit method on IndexeerService.
> 
> I'm not sure. Because then you would also need to add specific exception handling which is now part of the `addToSolrIndex` method.

No worries. I think I will need it in a next story but we will cross that bridge when we get to it :-)

Reply:
I will merge this PR now. We can continue the conversation elsewhere. :-)